awx/main/tests/functional/api/test_job_runtime_params.py

```diff
@@ -86,27 +87,20 @@ def test_job_ignore_unprompted_vars(runtime_data, job_template_prompts, post, ad
     mock_job = mocker.MagicMock(spec=Job, id=968, **runtime_data)
 
     with mocker.patch.object(JobTemplate, 'create_unified_job', return_value=mock_job):
-        with mocker.patch('awx.api.serializers.JobSerializer.to_representation'):
-            response = post(reverse('api:job_template_launch', kwargs={'pk':job_template.pk}),
-                            runtime_data, admin_user, expect=201)
-            assert JobTemplate.create_unified_job.called
-            assert JobTemplate.create_unified_job.call_args == ({'extra_vars':{}},)
-
-    # Check that job is serialized correctly
-    job_id = response.data['job']
-    assert job_id == 968
-
-    # If job is created with no arguments, it will inherit JT attributes
-    mock_job.signal_start.assert_called_once()
-
-    # Check that response tells us what things were ignored
-    assert 'job_launch_var' in response.data['ignored_fields']['extra_vars']
-    assert 'job_type' in response.data['ignored_fields']
-    assert 'limit' in response.data['ignored_fields']
-    assert 'inventory' in response.data['ignored_fields']
-    assert 'credential' in response.data['ignored_fields']
-    assert 'job_tags' in response.data['ignored_fields']
-    assert 'skip_tags' in response.data['ignored_fields']
+        response = post(reverse('api:job_template_launch', kwargs={'pk':job_template.pk}),
+                        runtime_data, admin_user, expect=400)
+        # Assert no job is created
+        assert not JobTemplate.create_unified_job.called
+        assert not mock_job.signal_start.called
+
+    # Check that response tells us what things were rejected
+    assert 'job_launch_var' in response.data['extra_vars'][0]
+    assert 'job_type' in response.data['job_type'][0]
```

Comment:
This is a strange looking response `'job_type' under a dictionary key labeled `job_type` ?

Reply:
Agree. Wording for the line here:

https://github.com/ansible/awx/pull/540/files#diff-0e240f76c0cd6b9b3b62205a03549d45R399

It would probably be just as good to say:

> Field is not configured to prompt on launch.

I would hope that the UI passthrough of the errors includes the key somewhere. I should check that.

Reply:
wait, nevermind, this isn't relevant to the UI, because they know what can and can't be provided, so they will never give the user the option.

Reply:
error text and assertion have been changed